File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode1957DeleteCharactersToMakeFancyString.cpp
#include "LeetCode1957DeleteCharactersToMakeFancyString.h"
using namespace std;

string LeetCode1957DeleteCharactersToMakeFancyString::makeFancyString(string s)
{
    string result{};
    int consecutiveCounter{};
    char currentChar = 0;
    for (auto c : s)
    {
        if (c != currentChar)
        {
            consecutiveCounter = 1;
            currentChar = c;
        }
        else
        {
            consecutiveCounter++;
        }
        if (consecutiveCounter < 3)
        {
            result += c;
        }
    }
    return result;
}

File: LeetCodeProblemsGo/LeetCode1957DeleteCharactersToMakeFancyString/LeetCode1957DeleteCharactersToMakeFancyString.go
package LeetCode1957DeleteCharactersToMakeFancyString

import "strings"

func makeFancyString(s string) string {
	var builder strings.Builder
	consecutiveCount := 0
	var currentChar rune = 0
	for _, char := range s {
		if char != currentChar {
			consecutiveCount = 1
			currentChar = char

		} else {
			consecutiveCount++
		}
		if consecutiveCount < 3 {
			builder.WriteString(string(char))
		}
	}
	return builder.String()
}
